src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,97 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Handle file I/O.
+ */
+public class Storage {
+    private final Path storagePath;
+    private List<String> taskLst;
+
+    /**
+     * Construct a Storage from a specified filepath.
+     * @param storagePath the specified path
+     */
+    public Storage(Path storagePath) {
+        this.storagePath = storagePath;
+        try {
+            Path parent = storagePath.getParent();
+            Files.createDirectories(parent);
+
+            if(!Files.exists(storagePath)) {
+                Files.createFile(storagePath);
+            }
+
+            taskLst = Files.readAllLines(storagePath);
+
+        } catch (IOException err) {
+            System.out.println("File read error: " + err.getMessage());
+        }
+    }
+
+    /**
+     * Read tasks stored in the file.
+     * @return the list of tasks read
+     */
+    public ArrayList<Task> read() {
+        ArrayList<Task> lst = new ArrayList<>();
+        for(String line: this.taskLst) {
+            char type = line.charAt(1);
+            String desc = line.substring(7);
+            int index;
+            String name, time;
+            char done = line.charAt(3);
+            switch (type) {
+            case 'T':
+                lst.add(new ToDo(desc));
+                break;
+            case 'D':
+                index = desc.indexOf('(');
+                name = desc.substring(0, index);
+                time = desc.substring(index + 5, desc.length() - 1);
+                lst.add(new Deadline(name, LocalDate.parse(time, DateTimeFormatter.ofPattern("MMM dd yyyy"))));
+                break;
+            case 'E':
+                index = desc.indexOf('(');
+                name = desc.substring(0, index);
+                time = desc.substring(index + 5, desc.length() - 1);
+                lst.add(new Event(name, LocalDate.parse(time, DateTimeFormatter.ofPattern("MMM dd yyyy"))));
```

Comment:
May want to use a static variable for "MMM dd yyyy"

https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers